lib/livebook/utils.ex

```diff
@@ -27,6 +27,50 @@ defmodule Livebook.Utils do
     :crypto.strong_rand_bytes(42) |> Base.url_encode64() |> String.to_atom()
   end
 
+  @doc """
+  Generates a random binary id that includes node information.
+
+  ## Format
+
+  The id is formed from the following binary parts:
+
+    * 16B - hashed node name
+    * 9B - random bytes
+
+  The binary is base32 encoded.
+  """
+  @spec random_node_aware_id() :: id()
+  def random_node_aware_id() do
+    node_part = node_hash(node())
+    random_part = :crypto.strong_rand_bytes(9)
+    binary = <<node_part::binary, random_part::binary>>
+    # 16B + 9B = 25B is suitable for base32 encoding without padding
+    Base.encode32(binary, case: :lower)
```

Comment:
You can also use uri encoded base64. We can encode more information this way, wdyt?

Reply:
Having `_-` in the url is less pretty, but in the end it's probably better to have the ids shorter, yeah :)

Reply:
Your call!

Reply:
Ah, but then we also have mixed upper and lower letters, gonna stick to base32 :see_no_evil: